homeassistant/components/smappee_official/sensor.py

```diff
@@ -0,0 +1,363 @@
+"""Support for monitoring a Smappee energy sensor."""
+import logging
+
+from homeassistant.const import DEVICE_CLASS_POWER, ENERGY_WATT_HOUR, POWER_WATT, VOLT
+from homeassistant.helpers.entity import Entity
+
+from .const import BASE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+TREND_SENSORS = {
+    "total_power": [
+        "Total consumption - Active power",
+        None,
+        POWER_WATT,
+        "total_power",
+        DEVICE_CLASS_POWER,
+    ],
+    "total_reactive_power": [
+        "Total consumption - Reactive power",
+        None,
+        POWER_WATT,
+        "total_reactive_power",
+        DEVICE_CLASS_POWER,
+    ],
+    "alwayson": [
+        "Always on - Active power",
+        "mdi:sleep",
+        POWER_WATT,
+        "alwayson",
+        None,
+    ],
+    "power_today": [
+        "Total consumption - Today",
+        "mdi:power-plug",
+        ENERGY_WATT_HOUR,
+        "power_today",
+        None,
+    ],
+    "power_current_hour": [
+        "Total consumption - Current hour",
+        "mdi:power-plug",
+        ENERGY_WATT_HOUR,
+        "power_current_hour",
+        None,
+    ],
+    "power_last_5_minutes": [
+        "Total consumption - Last 5 minutes",
+        "mdi:power-plug",
+        ENERGY_WATT_HOUR,
+        "power_last_5_minutes",
+        None,
+    ],
+    "alwayson_today": [
+        "Always on - Today",
+        "mdi:sleep",
+        ENERGY_WATT_HOUR,
+        "alwayson_today",
+        None,
+    ],
+}
+SOLAR_SENSORS = {
+    "solar_power": [
+        "Total production - Active power",
+        "mdi:white-balance-sunny",
```

Comment:
Please set the custom icon to `None` here when we set a device class.

Reply:
Same question as above, if we always need to use the power icon almost all our entities will have the lightning bolt icon (power). Wouldn't it look better though if we have some variety in the icons used?

Reply:
Icons are mainly a frontend concern. It's more important that the data is characterized properly.

Reply:
![always_on_icon](https://user-images.githubusercontent.com/58250533/84491314-a5882b00-aca4-11ea-8055-3c2da626c0f5.png)